draw-canvas.js

```diff
@@ -15,6 +15,11 @@ export default function main(pathname, backgroundImage) {
   const canv = createCanvas(width + 2 * PADDING, height + 2 * PADDING);
   const ctx = canv.getContext("2d");
 
+  /* When given a caller-provided image, we expect it to be given to 
+     us in a size appropriate to the grid they're creating, so we can
+     safely scale it. */
+  const fitImageToGrid = null != backgroundImage;
```

Comment:
I'm not sure about allowing scaling. It makes it very hard to line up a correctly created map. I tried creating a map with 40px grid and then tried to work out how wide and high to make the make. Takes a lot of trial and error. If we just go with clipping, a correctly created map will always line up.

Reply:
Playing around, I think drawing should be like this

```js
this.ctx.drawImage(img, 0, 0, this.width, this.height, this.padding, this.padding, this.width * this.zoom, this.height * this.zoom);
```

and then we should do some additional work to calculate the natural width and height of the image and set the canvas size to padding+natural width+padding and padding + natural height+padding if the height and width has not specifically been set using `10x10` etc in the URL.

Reply:
Good point. This'd also prevent us from needing the additional logic in draw(). Will address